src/components/modals/DateModal.tsx

```diff
@@ -1,79 +1,105 @@
 import DateTimePicker from '@react-native-community/datetimepicker'
 import * as React from 'react'
-import { useState } from 'react'
 import {
+  Appearance,
   Platform,
   Text,
   TextStyle,
-  TouchableOpacity,
-  useColorScheme
+  TouchableOpacity
 } from 'react-native'
 import { AirshipBridge, AirshipModal } from 'react-native-airship'
 
 import s from '../../common/locales/strings'
-import { useHandler } from '../../hooks/useHandler.js'
-import { useTheme } from '../services/ThemeContext'
+import { ThemeProps, withTheme } from '../services/ThemeContext'
 
-interface OwnProps {
+export interface Props {
   bridge: AirshipBridge<Date>
   initialValue: Date
 }
 
-type Props = OwnProps
+interface State {
+  darkMode: boolean
+  date: Date
+}
+
 /**
  * Shows the native iOS date picker inside a modal.
  */
+export class DateModalIos extends React.Component<Props & ThemeProps, State> {
+  subscription: { remove: () => void } | undefined
+
+  constructor(props: Props & ThemeProps) {
+    super(props)
+    this.state = {
+      darkMode: Appearance.getColorScheme() === 'dark',
+      date: props.initialValue
+    }
+  }
+
+  componentDidMount() {
+    this.subscription = {
+      // HACK: addChangeListener's type in RN 0.64 differs from the GUI's RN
+      // 0.71. Wrap to avoid errors for now...
+      remove: () =>
+        Appearance.addChangeListener(({ colorScheme }) => {
+          this.setState({ darkMode: colorScheme === 'dark' })
+        })
```

Comment:
Since React Native is just a devDependency, we should upgrade it.

I'm hoping it's just a package.json bump, but if it involves any extra changes, we could do it as the first commit of this PR.